lib/node.js

```diff
@@ -988,24 +993,41 @@ declare class stream$Duplex extends stream$Readable {
   constructor(options?: duplexStreamOptions): void;
   cork(): void;
   end(
-    chunk?: string | Buffer,
-    encodingOrCallback?: string | (data: any) => void,
-    callback?: (data: any) => void
+    chunk?: Buffer | string,
+    encodingOrCallback?: string | Function,
+    callback?: Function
   ): void;
   setDefaultEncoding(encoding: string): boolean;
   uncork() : void;
   write(
-    chunk?: string | Buffer,
-    encodingOrCallback?: string | (data: any) => void,
-    callback?: (data: any) => void
+    chunk: Buffer | string,
+    encodingOrCallback?: string | Function,
+    callback?: Function
+  ): boolean;
```

Comment:
The callback change here is actually a fix, because the callback is just a listener for the `flush` event, which has no documentation about any arguments:

1. https://nodejs.org/dist/latest-v5.x/docs/api/stream.html#stream_writable_write_chunk_encoding_callback